### PR TITLE
feat: copy as curl

### LINKED ISSUE #8789: Network Monitor -- Copy as CURL
We started it here, https://github.com/replayio/devtools/pull/7943

### PR DESCRIPTION
**Pull request by faga295**
closed #8789 
Kapture 2023-03-09 at 16 46 16

### CODE REVIEW

File: src/ui/components/NetworkMonitor/useNetworkContextMenu.tsx [hunk 1]
```diff
@@ -22,6 +22,10 @@ export default function useNetworkContextMenu(row: Row<RequestSummary>) {
     dispatch(setFocusRegionBeginTime(beginTime!, true));
   };
 
+  const copyAsCurl = () => {
+    navigator.clipboard.writeText(`curl '${row.values.url}'`);
```
Comment (jasonLaster) [inline src/ui/components/NetworkMonitor/useNetworkContextMenu.tsx:26]: I think we also want headers copied as well. Basically we want to be 1.1 with what chrome devtools does

**Comment by bvaughn** [comment]
Hey @faga295  :wave:

Be sure to check out our contributing guide before marking this "ready for review":
https://github.com/replayio/devtools/blob/main/docs/contributing.md#contributing-code

**Comment by faga295** [comment]
ok, I will follow that

**Comment by faga295** [comment]
> @faga295 Don't forget the description part of the guide:
> 
> > 1. Open a [PR on GitHub](https://github.com/replayio/devtools/pulls) with your changes. The PR description must include the following:
> >    
> >    * Link to the GitHub issue you are fixing (and any other relevant links)
> >    * Show how your change effects the UI/UX. (Screenshots, short Loom videos and Replays are good ways to show changes.)

thx, please check that this works as expected

File: src/ui/reducers/network.ts [hunk 2]
```diff
@@ -22,6 +23,7 @@ export type NetworkState = {
   requestBodies: Record<string, RequestBodyData[]>;
   requests: RequestInfo[];
   selectedRequestId: string | null;
+  copyCUrlAble: Record<RequestId, boolean>;
```
Comment (bvaughn) [inline src/ui/reducers/network.ts:26]: Why is this named "CUrlAble" ? Seems odd naming.

File: src/ui/components/NetworkMonitor/RequestRow.tsx [hunk 3]
```diff
@@ -38,8 +44,21 @@ export const RequestRow = ({
...
+
   const { contextMenu, onContextMenu } = useNetworkContextMenu(row);
 
+  const onNetworkContextMenu = useCallback(
+    async (event: any) => {
+      onContextMenu(event);
+      if (data.hasRequestBody) {
+        await client.Network.getRequestBody({ id: data.id, range: { end: 5e9 } }, sessionId);
+        dispath(enableCopyCUrl(data.id));
+      }
```
Comment (bvaughn) [inline src/ui/components/NetworkMonitor/RequestRow.tsx:57]: This looks unexpected to me.

Why would we need to eagerly fetch this? Why not just fetch it if/when the button is clicked?

**Review by bvaughn — COMMENTED** [review]
I don't think I understand why this feature was designed the way it was. It seems more complicated than I would have expected– pre-fetching and storing a "copyable" flag in Redux. Why do we need to do this?

**Comment by bvaughn** [comment]
Closing this in favor of #8885

File: src/ui/reducers/network.ts (hunk 2 above)
Comment (faga295) [reply src/ui/reducers/network.ts:26]: I will modify it to be more appropriate.

Comment (bvaughn) [reply src/ui/reducers/network.ts:26]: That's okay. No need.

**Pull request closed without merging**